### DISCUSSION TITLE
VSCode --> FM Script Workspace

### DISCUSSION THREAD

**Discussion by healthstreetnyc** [jwillinghalpern/filemaker-vscode-bundle#12] [Q&A]
Thanks for this awesome tool.  Quick questions:
1- is it possible to copy from .fmfn file in VS code to FM Script Workspace (via conversion to XML first)?
2- Is there a way to enable formatting ala Prettier?
Appreciate it.
Jared.

**Comment by jwillinghalpern** [comment]
Hey @healthstreetnyc those are both really good suggestions.
1. xml conversion. I do this using text expander in combination with this great applescript library: https://github.com/DanShockley/FmClipTools. My workflow is usually:
  1. copy something from FM, like layout objects or script steps
  2. convert to xml (simple key command on my computer)
  3. paste in vscode and edit
  4. convert to fm object (simple key command)
  5. paste back in fm.
  
I'd still consider adding that here because other people don't have the same tooling set up, so it could be really useful built-in. I'll make a note to look into that (and/or you are welcome to if you like that sort of thing!)


2- Is there a way to enable formatting ala Prettier?
  
This one feels trickier. I haven't researched much, but it might involve writing a language server, or better parsing rules for the calc language. This extension just does syntax highlighting because it's easily described in textmate syntax rules. Truth told, I don't know when I'll have time to look into adding this one. I'm happy to hear ideas!

But some people [have done this in the past](https://github.com/DonovanChan/Filemaker.tmbundle/blob/master/Commands/Tidy*.tmCommand). I think that adoptions for their formatter has been low because it doesn't offer many configuration options, and you might not agree with the formatting choices.

**Comment by jwillinghalpern** [comment]
I added an issue for the xml conversion here: https://github.com/jwillinghalpern/filemaker-vscode-bundle/issues/13
